### PR TITLE
ECL-496: Improved currency input validation

### PR DESCRIPTION
**Pull request by liandra-giacomine**

### CODE REVIEW

File: app/uk/gov/hmrc/economiccrimelevyreturns/forms/mappings/Formatters.scala [hunk 1]
```diff
@@ -24,6 +24,18 @@ import scala.util.control.Exception.nonFatalCatch
 
 trait Formatters {
 
+  private lazy val twoDecimalPattern = """^-?(\d*\.[0-9]{0,2})$""".r
```
Comment (TheDanielDoyle) [inline app/uk/gov/hmrc/economiccrimelevyreturns/forms/mappings/Formatters.scala:27]: Are we allowing negative numbers here?

Comment (TheDanielDoyle) [reply app/uk/gov/hmrc/economiccrimelevyreturns/forms/mappings/Formatters.scala:27]: The pattern passes with only a period and no decimal numbers

e.g. **1.**

File: test/uk/gov/hmrc/economiccrimelevyreturns/forms/behaviours/CurrencyFieldBehaviours.scala [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.economiccrimelevyreturns.forms.behaviours
+
+import play.api.data.{Form, FormError}
+
+trait CurrencyFieldBehaviours extends FieldBehaviours {
+
+  def currencyField(form: Form[_], fieldName: String, nonNumericError: FormError): Unit =
```
Comment (TheDanielDoyle) [inline test/uk/gov/hmrc/economiccrimelevyreturns/forms/behaviours/CurrencyFieldBehaviours.scala:23]: The following tests would be also useful.

Mixture of spaces before, beginning and after number/decimal, etc

File: app/uk/gov/hmrc/economiccrimelevyreturns/forms/mappings/Formatters.scala (hunk 1 above)
Comment (liandra-giacomine) [reply app/uk/gov/hmrc/economiccrimelevyreturns/forms/mappings/Formatters.scala:27]: Removing it

**Review by TheDanielDoyle — APPROVED** [review]

**Pull request merged**